main.tf

```diff
@@ -1,4 +1,61 @@
+resource "aws_elasticache_replication_group" "redis_no_destroy" {
```

Comment:
Why are we creating another resource here? Instead of adding the following block:
```
lifecycle {
    prevent_destroy = var.prevent_destroy
}
```
?